app/models/labware_creators/well_filter_allowing_partials.rb

```diff
@@ -15,7 +15,7 @@ def filter_by_state(requests)
   end
 
   def filter_requests(requests, well)
-    return nil if well.requests_as_source.empty?
```

Comment:
This has been looked at before by @andrewsparkes: https://github.com/sanger/limber/commit/e4d9e2f8f3de87e4d9dfa60bf79c8385087cd059. I don't think the existing behaviour is correct because this checks that the parent well contains the request (source of request is parent well). This means this can only be valid if the plate is the first plate in the pipeline (submitted on) and I don't think that is correct? I think it just happens that the other pipelines that use this meet that condition.

Reply:
Yeah we can discuss.
The labware creator was built for a specific plate at a branch point on the cherrypick plate in a specific pipeline, and works for that. It's not been designed to be generic.
I cannot remember why I had to change that line to get it to work though!
active_requests seems to use requests_as_source about 3 or 4 levels of method calls in. it's horribly complicated though.

Reply:
for chromium they would create the cherrypick, then do partial submissions on the wells in the cherrypick depending on which of 3 library prep types they wanted to do, hence partials. plus the same well could be submitted for more than one library prep. plus they might come back and repeat.
the well filter is having to filter on request type, library type and state to identify the right wells.

Reply:
the requests were not in the the aliquots on the cherrypick plate, because as you spotted that is the submission plate, they are in requests_as_source only.

Reply:
Yeah Katy and I discussed this yesterday and thought it best to create a new labware creator instead of modifying this because at the step we are interested in we aren't using partials so it doesnt make sense to use a partial stamp creator.